### PR TITLE
R doc 1360

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown [hunk 1]
```diff
@@ -0,0 +1,116 @@
+﻿# Querying : Graph Queries
+
+Graph queries are a new type of queries that allows you to query your documents as if they were a graph.
```
[the remark this answers is not in the document]
Comment (talweiss1982) [reply Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: okay ill come up with some samples

File: Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    ...
+    User: Users/1-A
+    ...
+}
+{CODE-BLOCK/}  
+- Properties that are nested objects that have a property that is a document key, the nested object is considered an edge in this case.  
+{CODE-BLOCK:plain}
+{
+    ...
+    User:
```
[the remark this answers is not in the document]
Comment (arekpalinski) [reply Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: @talweiss1982 please apply this change

[the remark this answers is not in the document]
Comment (talweiss1982) [reply Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: done

File: Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+![Figure 2. Simple pattern query result](images/basic_pattern_query.JPG "Simple pattern query result")
+
+As we can see the result of the above query is a tuple of an order as `o`, employee as `e` and the employee id as `edge`.  
+
+The result is simmilar to the following `RQL` query but not exactly:  
+
+{CODE-BLOCK:plain}
+from Orders as o 
+select o, o.Employee
```
Comment (ayende) [inline Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: To make it identical, you can use `load` here

File: Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+Lets query the employee:  
+
+{CODE-BLOCK:plain}
+match (Orders as o)-[Employee as edge]->(Employees as e where HiredAt.Year = 1993)
+{CODE-BLOCK/} 
+
+![Figure 3. Filter destination pattern query result](images/filtering_destination_pattern.JPG "Filter destination pattern query result")
+
+As you can see there are now `265` result rather than `830` as before since we filtered all tuples where their employee didn't match the query.
```
Comment (ayende) [inline Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: `265` results

File: Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+{CODE-BLOCK/} 
+
+![Figure 3. Filter destination pattern query result](images/filtering_destination_pattern.JPG "Filter destination pattern query result")
+
+As you can see there are now `265` result rather than `830` as before since we filtered all tuples where their employee didn't match the query.  
+
+We can concatenate more edges to the query, lets fetch the employee boss too:  
+
+{CODE-BLOCK:plain}
+match (Orders as o)-[Employee as _ ]->(Employees as e where HiredAt.Year = 1993)-[ReportsTo as __ ]->(Employees as boss)
```
Comment (ayende) [inline Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: `_` once should do it, no need for `__`

Comment (ayende) [reply Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: For that matter, you can not use any aliases here

File: Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+Nodes are documents, you select nodes using regular [RQL](../what-is-rql) queries inside a with clause.  
+
+{CODE-BLOCK:plain}
+with {from Orders} as o
+{CODE-BLOCK/}
+
+In the above with clause we choose nodes from the Orders collection and we will later refer to them as `o`.  
+
+### Edges
+Edges are properties of documents they could be:
```
Comment (ayende) [inline Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: You moved here from the Northwind model to some other model.
Let's keep using northwind here

File: Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+        "UserId":"users/283-A"
+        ...
+    }
+    ]
+    ...
+}
+{CODE-BLOCK/}  
+
+You choose edges using `with edges` clause e.g.  
+`with edges (Lines) { where Discount >= 0.25 select Product } as cheap`
```
Comment (ayende) [inline Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: You need to explain the shorthand `match (Orders as o)` vs. the `with { } as o`

File: Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+The above graph query over the `Northwind` database would return the tuples (o, p, cheap).  
+
+- o is an order document  
+- cheap is an order line with discount greater or equal 25%  
+- p is the product refered to by the cheap order line  
+
+### Implicit with clauses
+
+There is actually no need to define the `with` and `with edges` clauses for the above query, you can write a graph query like so:
```
Comment (ayende) [inline Documentation/4.2/Raven.Documentation.Pages/indexes/querying/graph/graph-queries.markdown]: There need to be better structure for this document. Maybe consult with @Danielle9897  / @reebhub
The information is here, but isn't presented in the proper order to clarify intent and make it easy to grok

**Pull request merged**